### PR TITLE
More typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
## Summary

Fix more typos.

About `convertor` → `converter`, see for example https://thecontentauthority.com/blog/convertor-vs-converter.

## Test Plan

—

### CODE REVIEW

File: crates/red_knot_python_semantic/src/types.rs [hunk 1]
```diff
@@ -899,7 +899,7 @@ impl<'db> Type<'db> {
                 Type::SubclassOf(_),
             ) => true,
             (Type::SubclassOf(_), _) | (_, Type::SubclassOf(_)) => {
-                // TODO: Once we have support for final classes, we can determine disjointness in some cases
+                // TODO: Once we have support for final classes, we can determine disjointedness in some cases
```
Comment (AlexWaygood) [inline crates/red_knot_python_semantic/src/types.rs:902]: same comment as last time :-)

https://github.com/astral-sh/ruff/pull/14863#discussion_r1875617796

**Review by AlexWaygood — COMMENTED** [review]
Thanks!

File: crates/red_knot_python_semantic/src/types.rs (hunk 1 above)
Comment (DimitriPapadopoulos) [reply crates/red_knot_python_semantic/src/types.rs:902]: There's no such word in the [OED](https://www.oed.com/search/dictionary/?q=disjointness). But then I guess the word is specifically used in a mathematical context.

Comment (AlexWaygood) [reply crates/red_knot_python_semantic/src/types.rs:902]: "disjointness" still sounds slightly better to me in this context. but whatever, I think the meaning is perfectly clear with either, and hopefully in the long run we'll just do the TODO anyway :-)

Comment (carljm) [reply crates/red_knot_python_semantic/src/types.rs:902]: I think "disjointedness" is a totally different word with a different meaning, and wouldn't have merged the PR with this change in it. But yeah, let's just do the TODO and get rid of the comment :)

Comment (DimitriPapadopoulos) [reply crates/red_knot_python_semantic/src/types.rs:902]: Indeed, `disjointness` appears to be used specifically in mathematics for disjoint sets.

Comment (DimitriPapadopoulos) [reply crates/red_knot_python_semantic/src/types.rs:902]: Reverted in #14906.

**Pull request merged**